From a1b374318864ff23d4354490ee24cf822a96f8ac Mon Sep 17 00:00:00 2001
From: Junxiao Chang <junxiao.chang@intel.com>
Date: Tue, 27 Apr 2021 13:51:34 +0800
Subject: [PATCH] xfrm: fix lock usage issue

in 5.10.30-rt37, there is a lock init issue which should be
introduced by conflict handling.

Signed-off-by: Junxiao Chang <junxiao.chang@intel.com>
---
 net/xfrm/xfrm_state.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/net/xfrm/xfrm_state.c b/net/xfrm/xfrm_state.c
index 7a2840d53654..77499abd9f99 100644
--- a/net/xfrm/xfrm_state.c
+++ b/net/xfrm/xfrm_state.c
@@ -2663,8 +2663,7 @@ int __net_init xfrm_state_init(struct net *net)
 	net->xfrm.state_num = 0;
 	INIT_WORK(&net->xfrm.state_hash_work, xfrm_hash_resize);
 	spin_lock_init(&net->xfrm.xfrm_state_lock);
-	seqcount_spinlock_init(&net->xfrm.xfrm_state_hash_generation,
-			       &net->xfrm.xfrm_state_lock);
+	seqcount_init(&net->xfrm.xfrm_state_hash_generation);
 	return 0;
 
 out_byspi:
-- 
2.25.1

